shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -367,6 +373,12 @@ - (void)setupChannels {
       binaryMessenger:self.binaryMessenger
                 codec:[FlutterJSONMethodCodec sharedInstance]]);
 
+  if ([_initialRoute length] > 0) {
+    // Flutter isn't ready to receive this method call yet but the channel buffer will cache this.
+    [_navigationChannel invokeMethod:@"setInitialRoute" arguments:_initialRoute];
```

Comment:
You should probably use the variant of invokeMethod that does error checking and print out if there is an error.

Reply:
oh true. I agree. Though it looks like we're using the non-callback variant across the embedding code currently. I'd like to keep it consistent and move all call sites if we do.